### DISCUSSION TITLE
What is the directory where Microsoft UI Xaml is installed on Windows 11?

### DISCUSSION THREAD

**Discussion by Dhyfer1** [microsoft/microsoft-ui-xaml#9861] [Q&A]
As the title says, and it is to know if after a reinstallation of Windows, the folder where it is installed remains.

**Comment by DarranRowe** [comment]
Honestly, the answer to this question isn't easy. It is easy to say yes, a copy of Microsoft.UI.Xaml.dll or the Microsoft.UI.Xaml package will be installed on a system after Windows 11 is reinstalled. But this is an incomplete answer and may be useless to you.
You really should provide more information as to why you are asking this, but in general, you shouldn't assume that any specific version will be available after Windows is reinstalled.

Although, to give a partial answer. The default package installation directory for Windows is %ProgramFiles%\WindowsApps. If there is going to be a version installed, it will be here somewhere.

**Reply by Dhyfer1** [reply]
Hi @DarranRowe 

Well, I have reinstalled Windows 11 from System>Recovery>Reset this PC. It was a local reinstall, clean without keeping files or apps. Before reinstalling I had Microsoft UI Xaml installed.

After the reinstallation and once my PC is ready, I download my favorite programs and also download the latest version of Microsoft UI Xaml from this repo, but when I want to install the appx package manually this message appears:

![Screenshot 2024-07-30 203949.png](https://github.com/user-attachments/assets/0cbbcba7-3794-4d49-83f0-ab093c3e4117)

So there are still traces of Microsoft Xaml UI after a reinstall, and I don't know how to close the conflicting app, MicrosoftDesktopAppInstaller, so that it can install Microsoft Xaml UI. 

By the way, I was looking %ProgramFiles%\WindowsApps and there is a folder called "Microsoft.UI.Xaml.CBS_8wekyb3d8bbwe". I think this folder that was there before reinstalling Windows survived the reinstall.

Edit: After restarting Windows, I still get the same message, I can't install (or reinstall?) Microsoft UI Xaml

**Reply by DarranRowe** [reply]
First, Microsoft.UI.Xaml.CBS has nothing to do with this. This is a private internal version of WinUI exclusively for use by Windows itself. So it isn't a surprise that it will survive a reinstall. You can tell this because it exists in the Windows directory itself, not WindowsApps.
Secondly, the reason why this is awkward is because several inbox applications use Microsoft.UI.Xaml 2.8. The fun thing about this is that Windows will always try to exit any applications that rely on a package, even if the operation on that package ends up as a noop. To show what I mean:

![Screenshot 2024-07-31 034817](https://github.com/user-attachments/assets/a8046d08-0220-4f76-9ed2-9ca7bd25b67f)

This is the version of Microsoft.UI.Xaml 2.8 installed on my system. If you also compare the version numbers, you should see that it is the same version as you are trying to work with. Anyway, the reason I am making this point is:

![Screenshot 2024-07-31 035035](https://github.com/user-attachments/assets/d27f79cf-dae8-4813-9586-9ac93beaf6d6)

Even if you are trying to install the version that is installed on your system, Windows will still try to exit all applications that are using the package. So, the first and biggest question that I have to ask, are you sure that you do not have the current version of Microsoft.UI.Xaml 2.8 installed on your system already?

On the topic of several inbox applications use Microsoft.UI.Xaml 2.8, the desktop app installer is the UI which you took the screenshot of. In other words, double clicking on a .appx or .msix file will run the desktop app installer, but the desktop app installer is what then tries to install Microsoft.UI.Xaml 2.8. In other words, you cannot install the x64 version Microsoft.UI.Xaml 2.8 by double clicking on it.
The only option you have is to use powershell to install it using add-appxpackage, but even then you are still going to have to use -DeferRegistrationWhenPackagesAreInUse or -ForceApplicationShutdown.

In other words, this error message is valid since several components that are installed along side Windows (like the game bar and terminal) use this version of Microsoft.UI.Xaml.

**Reply by Dhyfer1** [reply]
You're right, I was in the wrong folder. In my other comment the folder I was referring to is in: C:\ > Windows > System Apps folder, I'm sorry.

>So, the first and biggest question that I have to ask, are you sure that you do not have the current version of Microsoft.UI.Xaml 2.8 installed on your system already?

I don't know, but I was looking in %ProgramFiles%\Windows Apps and found 5 folders that start with microsoft.ui.xaml 

![Screenshot 2024-07-30 221717.png](https://github.com/user-attachments/assets/b90216f4-b8e3-42ab-b152-eb8ec714dd98)

>In other words, you cannot install the x64 version Microsoft.UI.Xaml 2.8 by double clicking on it.

You are wrong about that. I have installed the Microsoft UI Xaml appx file many times by double clicking on it and waiting for the bar to reach 100%, when it finishes it indicates that Microsoft UI xaml was installed successfully, this happens after a reinstall of Windows using a USB from the BIOS.

On the contrary, if I reinstall Windows from Windows settings and then try to install the Microsoft UI xaml appx, I get the error I showed you before.

Short summary:
1. Installing Windows with USB flash from BIOS
2. Installing the Microsoft UI xaml appx file manually by double clicking on it and waiting for it to finish installing
3. I use the PC for some time (days, weeks, months, etc.) and do a clean reinstall from Windows settings
4. After the reinstallation, I try to install the appx but I get the error 

I also wanted to show you that when I open the appx file, the task manager appears as App Installer, in other words, the app is creating a conflict in itself. You can also see the appx file window with the Install button, after clicking on it it reaches 1% installation and then the error appears

![Screenshot 2024-07-30 221114.png](https://github.com/user-attachments/assets/2ba87995-c5a9-435f-9b7e-542a4d5aa613)

This error, as I mentioned before, does not appear when I reinstall Windows from the BIOS with USB flash and installing the appx manually

**Reply by DarranRowe** [reply]
Well, you have the latest version of Microsoft.UI.Xaml 2.8 installed. Look at the directory, you will see 8.2310.310001.0 in the directory name.

>You are wrong about that. I have installed the Microsoft UI Xaml appx file many times by double clicking on it and waiting for the bar to reach 100%, when it finishes it indicates that Microsoft UI xaml was installed successfully, this happens after a reinstall of Windows using a USB from the BIOS.

I am not wrong about this, this is part of why I initially stated that the answer to this isn't easy and you shouldn't assume that any one particular version will be available after Windows is installed.
When you reset Windows, it will use the latest versions available from the Windows Update servers. This includes the latest patches for Windows and the latest versions of the inbox apps.
For your setup media, how old is it? If it wasn't created just before you installed it using the [Media Creation Tool](https://www.microsoft.com/en-us/software-download/windows11), then there can be old versions of the inbox apps that don't depend on 2.8. For example, the initial release of Windows 11 22H2 only came with Microsoft.UI.Xaml 2.1, 2.3, 2.4 and 2.7. The latest version, which is the version that your system would have reset to, also includes 2.8. While Microsoft themselves don't give access to this information, there are other resources that will show this information. But I think it is in bad taste to provide that information on a Microsoft controlled repository.

Another question. If you install from your USB installation media, but hold off installing Microsoft.UI.Xaml 2.8 until after Windows Updates have ran and you have updated all apps in the store, does it work then?

**Reply by Dhyfer1** [reply]
I do not use Media Creation Tool, I use Rufus instead. I have always used it with the latest ISO of the stable version of Windows 11 downloaded from the official Microsoft site, from the same link you posted.

> Another question. If you install from your USB installation media, but hold off installing Microsoft.UI.Xaml 2.8 until after Windows Updates have ran and you have updated all apps in the store, does it work then?

I listened to you. After reinstalling Windows 11 with USB flash from the BIOS, this is the version of Windows 11 without installing updates from Windows Update and app updates from the Microsoft Store. You can also see the Microsoft Xaml UI folders, this time there are 4 folders, I also downloaded the Microsoft UI xaml appx file, but I didn't use it until later.

![1.png](https://github.com/user-attachments/assets/e4b4974e-e59c-4430-b312-adaeda279541)

This screenshot shows the Windows version after installing the latest cumulative update, also all updates for Windows Security, Microsoft Defender, .Net Framework etc. were installed. I also installed the latest versions of App Installer, Notepad and Snipping Tool; no Microsoft Store update appeared because after upgrading to the latest stable version of Windows the store is already on its latest version.

![2.png](https://github.com/user-attachments/assets/4ee15021-ca67-4b3d-8eab-6f84479a0884)

At this point, now I really wanted to use the Microsoft UI Xaml appx file, so I opened it but it showed me the same message I showed in the first screenshot, I could not install it. 

But I would have been able to install it if I had not updated the App Installer app, believe me, I have done it several times. I can always install the appx file manually by double clicking on the file, then it shows an install bar reaching 100%, before installing app updates in the Store.

Now I know that the latest versions of Windows 11 already come with Microsoft UI Xaml 2.8, as I use it only for the TranslucentTB app, which in 21H2 and 22H2 indicated that Microsoft UI Xaml had to be installed in order to use it.

**Comment by driver1998** [comment]
Microsoft.DesktopAppInstaller is the package install UI you are seeing. 

The fact that DesktopAppInstaller works shows Microsoft.UI.Xaml.2.8 should already been installed and working.

If you still need to reinstall it, you may need to use other means to install the package instead of Microsoft.DesktopAppInstaller, like Add-AppxPackage in Powershell.

**Reply by Dhyfer1** [reply]
>you may need to use other means to install the package instead of Microsoft.DesktopAppInstaller, like Add-AppxPackage in Powershell.

And what is the method to install/uninstall Microsoft Xaml UI with PowerShell?
